components/PlantPreview.js

```diff
@@ -51,6 +62,12 @@ export default function PlantPreview({ plant }) {
         </StyledLink>
         <StyledBotanicalName>{plant.botanical_name}</StyledBotanicalName>
       </StyledInfo>
+      <ButtonAddPlant
+        OnToggleMyPlants={handleToggleMyPlants}
+        isMyPlant={isMyPlant}
+        plant={plant}
```

Comment:
Why do you pass the plant to the button?
The button only needs to know if he is favourite or not and a function what to do when clicked.